pkg/docker/types.go

```diff
@@ -0,0 +1,16 @@
+package docker
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type imageResolverFunc func(context.Context, string, config.DockerCredentials) (*ImageManifest, error)
+
+type ImageManifest struct {
+	// We only ever expect the digest to be the `algorithm:hash`
+	digest    string
```

Comment:
Final thought, are there any type strength advantages to using `Digest` object here?